backend/src/activitypub/ActivityPub.js

```diff
@@ -12,102 +12,99 @@ import request from 'request'
 import as from 'activitystrea.ms'
 import NitroDataSource from './NitroDataSource'
 import router from './routes'
-import dotenv from 'dotenv'
 import Collections from './Collections'
 import uuid from 'uuid/v4'
+import express from 'express'
+import dotenv from 'dotenv'
 const debug = require('debug')('ea')
 
 let activityPub = null
 
 export { activityPub }
 
-export default class ActivityPub {
-  constructor (activityPubEndpointUri, internalGraphQlUri) {
-    this.endpoint = activityPubEndpointUri
-    this.dataSource = new NitroDataSource(internalGraphQlUri)
-    this.collections = new Collections(this.dataSource)
-  }
-
-  static init (server) {
-    if (!activityPub) {
-      dotenv.config()
-      activityPub = new ActivityPub(process.env.CLIENT_URI || 'http://localhost:3000', process.env.GRAPHQL_URI || 'http://localhost:4000')
+export default function (server) {
+  if (!activityPub) {
+    activityPub = new ActivityPub()
 
-      // integrate into running graphql express server
+    // integrate into running express server
+    if (server) {
       server.express.set('ap', activityPub)
       server.express.use(router)
-      console.log('-> ActivityPub middleware added to the graphql express server')
-    } else {
-      console.log('-> ActivityPub middleware already added to the graphql express server')
+      console.log('-> ActivityPub middleware added to the express server')
+      return
     }
+
+    // start standalone express server
+    dotenv.config()
+
+    const app = express()
+    app.set('ap', activityPub)
+    app.use(router)
+    app.listen(process.env.ACTIVITYPUB_PORT || 4010, () => {
```

Comment:
> [<img alt="roschaefer" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/roschaefer) **Authored by [roschaefer](https://github.com/roschaefer)**
_<time datetime="2019-05-06T14:35:36Z" title="Monday, May 6th 2019, 4:35:36 pm +02:00">May 6, 2019</time>_

---
> **Outdated (history rewrite)** - original diff
---
```diff
@@ -12,102 +12,99 @@ import request from 'request'
 import as from 'activitystrea.ms'
 import NitroDataSource from './NitroDataSource'
 import router from './routes'
-import dotenv from 'dotenv'
 import Collections from './Collections'
 import uuid from 'uuid/v4'
+import express from 'express'
+import dotenv from 'dotenv'
 const debug = require('debug')('ea')
 
 let activityPub = null
 
 export { activityPub }
 
-export default class ActivityPub {
-  constructor (activityPubEndpointUri, internalGraphQlUri) {
-    this.endpoint = activityPubEndpointUri
-    this.dataSource = new NitroDataSource(internalGraphQlUri)
-    this.collections = new Collections(this.dataSource)
-  }
-
-  static init (server) {
-    if (!activityPub) {
-      dotenv.config()
-      activityPub = new ActivityPub(process.env.CLIENT_URI || 'http://localhost:3000', process.env.GRAPHQL_URI || 'http://localhost:4000')
+export default function (server) {
+  if (!activityPub) {
+    activityPub = new ActivityPub()
 
-      // integrate into running graphql express server
+    // integrate into running express server
+    if (server) {
       server.express.set('ap', activityPub)
       server.express.use(router)
-      console.log('-> ActivityPub middleware added to the graphql express server')
-    } else {
-      console.log('-> ActivityPub middleware already added to the graphql express server')
+      console.log('-> ActivityPub middleware added to the express server')
+      return
     }
+
+    // start standalone express server
+    dotenv.config()
+
+    const app = express()
+    app.set('ap', activityPub)
+    app.use(router)
+    app.listen(process.env.ACTIVITYPUB_PORT || 4010, () => {
```


:thinking: what's the motivation for this extra port?

Reply:
> [<img alt="Mastercuber" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/Mastercuber) **Authored by [Mastercuber](https://github.com/Mastercuber)**
_<time datetime="2019-05-10T20:27:35Z" title="Friday, May 10th 2019, 10:27:35 pm +02:00">May 10, 2019</time>_

---


problems while trying out something. Would change it back to 4000 if necessary.

Reply:
> [<img alt="Mastercuber" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/Mastercuber) **Authored by [Mastercuber](https://github.com/Mastercuber)**
_<time datetime="2019-05-10T20:27:35Z" title="Friday, May 10th 2019, 10:27:35 pm +02:00">May 10, 2019</time>_

---


problems while trying out something. Would change it back to 4000 if necessary.